.github: Zip test reports before uploading

seemethere opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#60475 .github: Zip test reports before uploading**

Uploading many artifacts can cause issues with GHA backend leading to
errors on our side. To be safe let's zip our artifacts into one archive
so that we avoid uploading too many files at once.

See: https://github.com/actions/upload-artifact#too-many-uploads-resulting-in-429-responses

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D29307205](https://our.internmc.facebook.com/intern/diff/D29307205)

seemethere committed:
.github: Zip test reports before uploading

Uploading many artifacts can cause issues with GHA backend leading to
errors on our side. To be safe let's zip our artifacts into one archive
so that we avoid uploading too many files at once.

See: https://github.com/actions/upload-artifact#too-many-uploads-resulting-in-429-responses

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".github: Zip test reports before uploading"


Uploading many artifacts can cause issues with GHA backend leading to
errors on our side. To be safe let's zip our artifacts into one archive
so that we avoid uploading too many files at once.

See: https://github.com/actions/upload-artifact#too-many-uploads-resulting-in-429-responses

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

samestep commented on .github/templates/linux_ci_workflow.yml.j2:
```diff
@@ -306,6 +306,12 @@ jobs:
+        run: |
+          # Remove any previous test reports if they exist
+          rm -f test-reports-*.zip
+          zip -r "test-reports-${TEST_CONFIG}.zip" test/test-reports
```
judging from #58250 it seems like this is wrong?

seemethere replied:
Updated

seemethere committed:
Update on ".github: Zip test reports before uploading"


Uploading many artifacts can cause issues with GHA backend leading to
errors on our side. To be safe let's zip our artifacts into one archive
so that we avoid uploading too many files at once.

See: https://github.com/actions/upload-artifact#too-many-uploads-resulting-in-429-responses

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

samestep reviewed (approved):
lgtm if CI passes

seemethere commented:
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29307205).

seemethere committed:
Update on ".github: Zip test reports before uploading"


Uploading many artifacts can cause issues with GHA backend leading to
errors on our side. To be safe let's zip our artifacts into one archive
so that we avoid uploading too many files at once.

See: https://github.com/actions/upload-artifact#too-many-uploads-resulting-in-429-responses

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D29307205](https://our.internmc.facebook.com/intern/diff/D29307205)

[ghstack-poisoned]

walterddr reviewed (approved):
this is nice. but does it work well with test1/test2? (I think we previously establish that b/c the test files generated from the 2 shards are unique, thus uploading them automatically merges.

walterddr commented on .github/templates/linux_ci_workflow.yml.j2:
```diff
@@ -306,6 +306,12 @@ jobs:
+        run: |
+          # Remove any previous test reports if they exist
+          rm -f test-reports-*.zip
+          zip -r "test-reports-${TEST_CONFIG}.zip" test -i '*.xml'
```
qq: will `tar` run faster? (size is not the issue here, but # of small files right?)

seemethere replied:
Potentially tar could run faster, just using zip here since zip is what were using in previous steps for artifacts

seemethere commented:
> this is nice. but does it work well with test1/test2? (I think we previously establish that b/c the test files generated from the 2 shards are unique, thus uploading them automatically merges.

It should be fine since we're uploading separate zips based on the `TEST_CONFIG` variable, so ideally they should come out like:
- test-report-1.zip
- test-report-2.zip

seemethere committed:
Update on ".github: Zip test reports before uploading"


Uploading many artifacts can cause issues with GHA backend leading to
errors on our side. To be safe let's zip our artifacts into one archive
so that we avoid uploading too many files at once.

See: https://github.com/actions/upload-artifact#too-many-uploads-resulting-in-429-responses

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D29307205](https://our.internmc.facebook.com/intern/diff/D29307205)

[ghstack-poisoned]

seemethere committed:
Update on ".github: Zip test reports before uploading"


Uploading many artifacts can cause issues with GHA backend leading to
errors on our side. To be safe let's zip our artifacts into one archive
so that we avoid uploading too many files at once.

See: https://github.com/actions/upload-artifact#too-many-uploads-resulting-in-429-responses

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D29307205](https://our.internmc.facebook.com/intern/diff/D29307205)

[ghstack-poisoned]

seemethere commented:
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29307205).

seemethere committed:
Update on ".github: Zip test reports before uploading"


Uploading many artifacts can cause issues with GHA backend leading to
errors on our side. To be safe let's zip our artifacts into one archive
so that we avoid uploading too many files at once.

See: https://github.com/actions/upload-artifact#too-many-uploads-resulting-in-429-responses

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D29307205](https://our.internmc.facebook.com/intern/diff/D29307205)

[ghstack-poisoned]

seemethere committed:
Update on ".github: Zip test reports before uploading"


Uploading many artifacts can cause issues with GHA backend leading to
errors on our side. To be safe let's zip our artifacts into one archive
so that we avoid uploading too many files at once.

See: https://github.com/actions/upload-artifact#too-many-uploads-resulting-in-429-responses

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D29307205](https://our.internmc.facebook.com/intern/diff/D29307205)

[ghstack-poisoned]

seemethere commented:
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29307205).

seemethere committed:
Update on ".github: Zip test reports before uploading"


Uploading many artifacts can cause issues with GHA backend leading to
errors on our side. To be safe let's zip our artifacts into one archive
so that we avoid uploading too many files at once.

See: https://github.com/actions/upload-artifact#too-many-uploads-resulting-in-429-responses

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D29307205](https://our.internmc.facebook.com/intern/diff/D29307205)

[ghstack-poisoned]

seemethere commented:
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29307205).

seemethere committed:
Update on ".github: Zip test reports before uploading"


Uploading many artifacts can cause issues with GHA backend leading to
errors on our side. To be safe let's zip our artifacts into one archive
so that we avoid uploading too many files at once.

See: https://github.com/actions/upload-artifact#too-many-uploads-resulting-in-429-responses

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D29307205](https://our.internmc.facebook.com/intern/diff/D29307205)

[ghstack-poisoned]

seemethere committed:
Update on ".github: Zip test reports before uploading"


Uploading many artifacts can cause issues with GHA backend leading to
errors on our side. To be safe let's zip our artifacts into one archive
so that we avoid uploading too many files at once.

See: https://github.com/actions/upload-artifact#too-many-uploads-resulting-in-429-responses

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D29307205](https://our.internmc.facebook.com/intern/diff/D29307205)

[ghstack-poisoned]

seemethere committed:
Update on ".github: Zip test reports before uploading"


Uploading many artifacts can cause issues with GHA backend leading to
errors on our side. To be safe let's zip our artifacts into one archive
so that we avoid uploading too many files at once.

See: https://github.com/actions/upload-artifact#too-many-uploads-resulting-in-429-responses

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D29307205](https://our.internmc.facebook.com/intern/diff/D29307205)

[ghstack-poisoned]

seemethere commented:
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29307205).
